packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -180,6 +180,16 @@ export default createRule<Options, MessageIds>({
           checkExpression(expression);
         }
       },
+      CallExpression(node: TSESTree.CallExpression): void {
+        if (
+          node.callee.type === AST_NODE_TYPES.Identifier &&
+          // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+          node.callee.name === 'String' &&
```

Comment:
```ts
// an annoying thing people sometimes do
function String(value) {
  try { return JSON.stringify(value); }
  catch { return `${value}`; }
}

// ...

declare const myValue: object;
String(myValue);
```

This should check with the scope analyzer to make sure the node is actually the built-in `String`, I think.

Reply:
@JoshuaKGoldberg Thanks! [addf652](https://github.com/typescript-eslint/typescript-eslint/pull/10005/commits/addf65281b9dcc071471a4d0f4fab58b99f9495d)